src/Employer/ESFA.DAS.EmployerFeedbackEmail.Data/EmployerFeedbackRepository.cs

```diff
@@ -105,17 +105,47 @@ public async Task InsertNewSurveyInviteCodes(IEnumerable<EmployerSurveyInvite> n
                 {
                     UniqueSurveyCode = Guid.NewGuid(),
                     x.UserRef, 
-                    x.Ukprn
+                    x.Ukprn,
+                    x.AccountId
                 };
             });
 
             var sql = $@"
                         INSERT INTO EmployerSurveyCodes
-                        VALUES(@UniqueSurveyCode, @UserRef, @Ukprn, NULL)";
+                        VALUES(@UniqueSurveyCode, @UserRef, @Ukprn, @AccountId, NULL)";
 
             await _dbConnection.ExecuteAsync(sql, newCodesToCreate);
         }
 
+        public async Task<Guid> GetOrCreateSurveyCode(Guid UserRefParam, long UkprnParam, long AccountIdParam)
+        {
+            var sql = $@"
+                        SELECT UniqueSurveyCode FROM EmployerSurveyCodes
+                        WHERE UserRef = @UserRefParam AND Ukprn = @UkprnParam AND AccountId = @AccountIdParam";
+            var result = await _dbConnection.QueryAsync<Guid>(sql,new{UserRefParam,UkprnParam, AccountIdParam });
+            if (result.Count() == 0)
```

Comment:
This could be better expressed as...
```suggestion
            if (!result.Any())
```

Reply:
Done - Always forget this is a thing